Tasks/Common/nuget-task-common/NuGetToolRunner.ts

```diff
@@ -155,6 +156,78 @@ export function locateNuGetExe(userNuGetExePath: string): string {
     return toolPath;
 }
 
+function isHosted(): boolean {
+    // not an ideal way to detect hosted, but there isn't a variable for it, and we can't make network calls from here
+    // due to proxy issues.
+    const collectionUri = tl.getVariable("System.TeamFoundationCollectionUri");
+    return /\.visualstudio\.com$/i.test(collectionUri);
```

Comment:
nit for later - there's other hosted urls for PPE, bashing, etc...  we can follow up on a more reliable way later (variable in job message)